src/test/fuzz/asmap_direct.cpp

```diff
@@ -6,41 +6,40 @@
 #include <test/fuzz/fuzz.h>
 
 #include <cstdint>
+#include <optional>
 #include <vector>
 
 #include <assert.h>
 
 void test_one_input(const std::vector<uint8_t>& buffer)
 {
     // Encoding: [asmap using 1 bit / byte] 0xFF [addr using 1 bit / byte]
-    bool have_sep = false;
-    size_t sep_pos;
+    std::optional<size_t> sep_pos;
     for (size_t pos = 0; pos < buffer.size(); ++pos) {
         uint8_t x = buffer[pos];
         if ((x & 0xFE) == 0) continue;
         if (x == 0xFF) {
-            if (have_sep) return;
-            have_sep = true;
+            if (sep_pos) return;
             sep_pos = pos;
         } else {
             return;
         }
     }
-    if (!have_sep) return; // Needs exactly 1 separator
-    if (buffer.size() - sep_pos - 1 > 128) return; // At most 128 bits in IP address
+    if (!sep_pos) return; // Needs exactly 1 separator
+    if (buffer.size() - sep_pos.value() - 1 > 128) return; // At most 128 bits in IP address
```

Comment:
for a smaller diff you could rename the optional to `sep_pos_opt` and insert before this line

```cpp
const size_t sep_pos{sep_pos_opt.value()};